src/main/java/magus/task/TaskFinder.java

```diff
@@ -0,0 +1,218 @@
+package magus.task;
+
+import magus.console.Console;
+import magus.console.Parser;
+import magus.exception.ArgumentNotFoundException;
+import magus.exception.UnknownArgumentException;
+import magus.task.variant.Deadline;
+import magus.task.variant.Event;
+import magus.task.variant.Todo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class TaskFinder {
+    private final List<Task> taskList;
+
+    public TaskFinder(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    public List<Task> filterTasks(Parser parser) {
+        Parser subparser = new Parser(parser.getAdditionalInput());
+        String taskTypeString = subparser.getCommandCandidate();
+        TaskType taskType = TaskType.getEnum(taskTypeString);
+
+        List<Task> filteredTaskList = null;
+        switch (taskType) {
+        case TODO:
+            try {
+                filteredTaskList = filterTodos(subparser);
+            } catch (ArgumentNotFoundException ignored) {
+                // break out and try filter by description
+                break;
```

Comment:
is there a better way to do this rather than try--catch?
in general, it is not a good practice to have empty catch blocks

Reply:
Resolved in 1c307c3